homeassistant/components/recorder/__init__.py

```diff
@@ -273,14 +273,18 @@ def do_purge(now):
                     self.queue.task_done()
                     continue
 
-            with session_scope(session=self.get_session()) as session:
-                dbevent = Events.from_event(event)
-                session.add(dbevent)
-
-                if event.event_type == EVENT_STATE_CHANGED:
-                    dbstate = States.from_event(event)
-                    dbstate.event_id = dbevent.event_id
-                    session.add(dbstate)
+            try:
+                with session_scope(session=self.get_session()) as session:
+                    dbevent = Events.from_event(event)
+                    session.add(dbevent)
+
+                    if event.event_type == EVENT_STATE_CHANGED:
+                        dbstate = States.from_event(event)
+                        dbstate.event_id = dbevent.event_id
+                        session.add(dbstate)
+            except Exception as err:
```

Comment:
To not mark as saved, I can skip calling self.queue.task_done(), but will the item still be cleanly removed from the queue? I want to avoid the queue filling up in the event that a database is gone. 

I can move this into the try block

Other items implemented as requested